RileyLinkKitTests/PumpOpsSynchronousTests.swift

```diff
@@ -73,202 +82,351 @@ class PumpOpsSynchronousTests: XCTestCase {
     
     func testMultipleBolusEventsContainsFirstBolus() {
-        loadTestWithPumpModel(.Model522)
+        setUpTestWithPumpModel(.Model522)
         
         let events = [createSquareBolusEvent2010(), createBolusEvent2009()]
         
-        let (timeStampedEvents, _) = sut.convertPumpEventToTimestampedEvents(pumpEvents: events, startDate: Date.distantPast, pumpModel: pumpModel)
+        let (timeStampedEvents, _, _) = sut.convertPumpEventToTimestampedEvents(pumpEvents: events, startDate: Date.distantPast, pumpModel: pumpModel)
         
         assertArray(timeStampedEvents, containsPumpEvent: createSquareBolusEvent2010())
     }
     
     func testMultipleBolusEventsContainsSecondBolus() {
-        loadTestWithPumpModel(.Model522)
+        setUpTestWithPumpModel(.Model522)
         
         let events = [createSquareBolusEvent2010(), createBolusEvent2009()]
         
-        let (timeStampedEvents, _) = sut.convertPumpEventToTimestampedEvents(pumpEvents: events, startDate: Date.distantPast, pumpModel: pumpModel)
+        let (timeStampedEvents, _, _) = sut.convertPumpEventToTimestampedEvents(pumpEvents: events, startDate: Date.distantPast, pumpModel: pumpModel)
         
         assertArray(timeStampedEvents, containsPumpEvent: createBolusEvent2009())
     }
     
-    func testNonMutableSquareWaveBolusFor522() {
+    func testMultipleBolusEventsWith523ContainsLastEvent() {
+        setUpTestWithPumpModel(.Model523)
+        
+        let events = [createSquareBolusEvent2010(), createBolusEvent2009()]
+        
+        let (timestampedEvents, _, _) = sut.convertPumpEventToTimestampedEvents(pumpEvents: events, startDate: Date.distantPast, pumpModel: pumpModel)
+        
+        assertArray(timestampedEvents, containsPumpEvent: createBolusEvent2009())
+    }
+    
+    func testNonMutableSquareWaveBolusFor522IsReturned() {
         // device that can have out of order events
-        loadTestWithPumpModel(.Model522)
+        setUpTestWithPumpModel(.Model522)
         // 2009-07-31 09:00:00 +0000
         // 120 minute duration
         let squareWaveBolus = BolusNormalPumpEvent(availableData: Data(hexadecimalString: "010080048000240009a24a1510")!, pumpModel: pumpModel)!
         
         let events:[PumpEvent] = [squareWaveBolus]
         
-        let (timeStampedEvents, _) = sut.convertPumpEventToTimestampedEvents(pumpEvents: events, startDate: Date.distantPast, pumpModel: pumpModel)
+        let (timeStampedEvents, _, _) = sut.convertPumpEventToTimestampedEvents(pumpEvents: events, startDate: Date.distantPast, pumpModel: pumpModel)
         
         // It should be included
         XCTAssertTrue(array(timeStampedEvents, containsPumpEvent: squareWaveBolus))
     }
+    
+    func testEventAfterDelayedAppendEventFor522IsReturned() {
+        setUpTestWithPumpModel(.Model522)
+        
+        let tempEventBasal = createTempEventBasal2016()
+        let events:[PumpEvent] = [createSquareBolusEvent2010(), tempEventBasal]
```

Comment:
The order here is reversed (typically newer events are before older events in the list passed to convertPumpEventToTimestampedEvents()).  This is actually a pump date discontinuity that is being set up, but we're not detecting it because we skip the square wave bolus event.